### PR TITLE
Add unit tests #629

### LINKED ISSUE #629: Improve unit test coverage
### TODO
* All game object classes should be tested.
* Ideally all page state classes should also be tested.
* If there is time it would be good to test the vue components.

### PR DESCRIPTION
**Pull request by strinsberg**
Fixes #629

#### Overview of changes
- Added unit tests for all of the game object classes
- Made a few small updates to fix bugs that were found during the testing

Reviewer: @johnanvik

### CODE REVIEW

**Comment by strinsberg** [comment]
I ran into a problem with AI and playing SCAN card here, so Don't merge this until I can push a fix for it.

**Comment by strinsberg** [comment]
Problem was not that hard to track down and is now fixed.

File: src/classes/statusEffect/EffectWithCard.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ export default class EffectWithCard extends StatusEffect {
   }
 
   destroy () {
+    super.destroy()
```
Comment (johnanvik) [inline src/classes/statusEffect/EffectWithCard.js:10]: When destroying is not enough, you "super destroy"!

**Review by johnanvik — APPROVED** [review]
I'm assuming that the issue caught by GitHub is a false positive.

**Pull request merged**